### PR TITLE
KYLO-1460 Correctly exit spark Validator application

### PR DESCRIPTION
**Pull request by fabiannecci**
Stop the spark context with sparkContext.stop() and throw exception if the job fails

### CODE REVIEW

File: integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java [hunk 1]
```diff
@@ -45,57 +45,57 @@ public static void main(String[] args) {
...
-
-        try {
-            final int result = new Validator().run(System.out, args);
-            System.exit(result);
-        } catch (final Exception e) {
-            log.error("Failed to perform validation: ", e.toString(), e);
-            System.out.println(e);
-            System.exit(1);
-        }
+        new Validator().run(System.out, args);
```
Comment (KyloTech) [inline integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java]: MAJOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java [hunk 2]
```diff
@@ -9,9 +9,9 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (KyloTech) [inline integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java [hunk 3]
```diff
@@ -45,57 +45,57 @@ public static void main(String[] args) {
...
+                                                                  policyMap, hiveContext);
+
+            log.info("Persistence level: {}", params.getStorageLevel());
+            results.persist(StorageLevel.fromString(params.getStorageLevel()));
+
+            app.saveInvalidToTable(params.getTargetDatabase(), params.getInvalidTableName(), results, hiveContext);
+            app.saveValidToTable(params.getTargetDatabase(), params.getFeedTableName(), params.getValidTableName(), results, hiveContext);
+            app.saveProfileToTable(params.getTargetDatabase(), params.getProfileTableName(), params.getPartition(), results, hiveContext);
+            results.unpersist();
+            log.info("Validator app finished");
```
Comment (KyloTech) [inline integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java]: MAJOR String contains no format specifiers. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

**Comment by KyloTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java [hunk 4]
```diff
@@ -45,57 +45,57 @@ public static void main(String[] args) {
...
+            sparkContext.stop();
```
Comment (freastro) [inline integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java]: Please move this to a `finally` block. It'll make the code cleaner and prevent bugs if new catch blocks are added.

**Comment by freastro** [comment]
Pulled in manually.

**Pull request closed without merging**